pkg/skaffold/schema/validation/validation.go

```diff
@@ -87,6 +88,7 @@ func ProcessToErrorWithLocation(configs parser.SkaffoldConfigSet, validateConfig
 		errs = append(errs, validateCustomTest(config, config.Test)...)
 		errs = append(errs, validateGCBConfig(config, config.Build)...)
 		errs = append(errs, validateVerifyTests(config, config.Verify)...)
+		errs = append(errs, validateKptDeployerVersion(config, config.Deploy)...)
```

Comment:
+1, this is the better place to do this (vs where Helm does this).  One issue might be that currently this will disallow something like:
`skaffold render` + kpt deployer exists but this would actually work properly (only `dev`, `deploy`, etc. would actually give us an issue)